spec/requests/controllers/rooms_controller_spec.rb

```diff
@@ -7,21 +7,36 @@
     post '/validate_beta_user', params: { beta_code: 'correct-code' }
   end
 
-  context "user is authenticated" do
-    before(:each) { sign_in create(:user) }
+  describe 'GET #show' do
+    let(:room) { create(:room) }
 
-    describe 'GET #show' do
-      let(:room) { create(:room) }
+    it 'renders a room page' do
+      get room_path(room.public_id)
+      expect(response).to be_successful
+    end
 
-      it 'renders a room page' do
-        get room_path(room.public_id)
-        expect(response).to be_successful
-      end
+    it "doesn't allow routing to a room via id" do
+      expect do
+        get "/rooms/#{room.id}"
+      end.to raise_error(ActiveRecord::RecordNotFound)
+    end
+  end
 
-      it "doesn't allow routing to a room via id" do
+  context "user is authenticated" do
```

Comment:
Let's extract auth requirement into a shared examples so we can just mark any action we care with it instead of defining contexts. so, this would look like:
```
    describe 'POST #create' do
        it_behaves_like 'Auth Required'
```